### DISCUSSION TITLE
Question about Configuring Character-Specific AI Models in SillyTavern

### DISCUSSION THREAD

**Discussion by davostros** [SillyTavern/SillyTavern#3042] [Q&A]
Hello,

I'm trying to configure character-specific AI models in SillyTavern. My goal is to assign a different AI model to each character (e.g., LFM 40B for strategy, Gemini Flash for content creation, and Hermes for validation), using OpenRouter API.

However, when I open the character's profile settings, I don't see any field or option labeled "Model associated," "Backend," or "AI Model" to directly link the AI model to a specific character. I’ve checked the API connections and created profiles, but I can't seem to associate them with individual characters.

Could you provide a step-by-step guide or clarify how to link each character to its designated AI model within SillyTavern? Any advice or documentation reference would be greatly appreciated.

Thank you!

**Comment by dar4545** [comment]
Same question here. From what I understand, the connection profile is a global setting that can only be applied to all characters universally.

I’d love to see future updates allow associating different connection profiles with specific characters, especially for group chats.
This could provide finer control over API costs by assigning different roles (e.g., translation, narration, summarization) based on each model’s strengths.

Edit:
I found the dev has mentioned it in the similar request #2476. It seems unlikely that it can be implemented with the current architecture.

**Comment by CATIOR** [comment]
https://github.com/CATIOR/ST-Multi-Model-Chat/

Can it help?

**Comment by kinthaiofficial** [comment]
Character-specific model configuration is a good instinct — different character archetypes benefit from different model characteristics. A character that's supposed to be deeply knowledgeable on a domain benefits from a larger model; a character that's supposed to be quick and reactive benefits from a smaller, faster model.

A few things that work well for per-character model routing:

**Character capability manifest** — define per-character model preferences declaratively: `{"preferred_tier": "mid", "max_cost_per_message_cents": 0.5, "personality_requires": "creative"}`. The routing layer picks the cheapest model that satisfies the constraints.

**Personality-aware model selection** — creative/expressive characters benefit from higher temperature + larger models. Factual/knowledgeable characters benefit from lower temperature + retrieval augmentation. Knowledge-domain characters can use smaller models if fine-tuned for the domain.

**Context-length as a character attribute** — characters with long memory (ongoing relationship roleplay) need longer context windows. Characters for episodic interactions can use shorter context, which is cheaper.

**Cost visibility per character** — if you're running many characters, knowing "this character costs 3x more per message than that one" helps with character library curation.

For the platform design perspective, per-character model configuration is exactly the kind of economic decision that should be surfaced to users — "this character costs $0.02/message because it uses a larger model; this one costs $0.003/message." We think about this for KinthAI's agents too: https://blog.kinthai.ai/why-character-ai-forgets-you-persistent-memory-architecture — character persistence and memory design.

Is the model selection primarily for quality differentiation, or for cost control?
